core/parameters.class.inc.php

```diff
@@ -55,19 +55,19 @@ protected function ToXML(DOMNode $oRoot, $data = null, $sNodeName = null)
 		{
 			$data = $this->aData;
 		}
-				
-		if (is_array($data))
```

Comment:
Woops sorry, second thought... 
This if has moved, so now if `$data` isn't an array then now we will create an element... Is this change really intended ?

Reply:
Yes, it's moved to line 69 and 70.

Reply:
This change the code behavior... So I'm worried about regressions... Can you explain why did you move this line ?

Reply:
Because the element needs to be created also in the case it's not an array.

Reply:
Oh ok, I didn't noticed the modifications in the else block ! Before we had a createElement() call, now this is always done above and we're adding a textnode as a child. Sorry.

In theory ok for me so I'm closing this conversation. But... this is a supplementary reason to add a test. This PR won't be accepted without it.

Reply:
Great, still need to look into that. But you know, time for this is limited.

Reply:
Yes I understand.
If I ever find some time to write the test myself, did you checked the "allow maintainers" checkbox on the PR ?

Reply:
Yes I believe I did